pixeltable/func/tools.py

```diff
@@ -48,22 +48,27 @@ def ser_model(self) -> dict[str, Any]:
             'additionalProperties': False,  # TODO Handle kwargs?
         }
 
-    # `tool_calls` must be in standardized tool invocation format:
-    # {tool_name: {'args': {name1: value1, name2: value2, ...}}, ...}
-    def invoke(self, tool_calls: 'exprs.Expr') -> 'exprs.FunctionCall':
-        kwargs = {param.name: self.__extract_tool_arg(param, tool_calls) for param in self.parameters.values()}
-        return self.fn(**kwargs)
+    # The output of `tool_calls` must be a dict in standardized tool invocation format:
+    # {tool_name: [{'args': {name1: value1, name2: value2, ...}}, ...], ...}
+    def invoke(self, tool_calls: 'exprs.Expr') -> 'exprs.Expr':
+        from ..exprs import RELATIVE_PATH_ROOT as R
 
-    def __extract_tool_arg(self, param: Parameter, tool_calls: 'exprs.Expr') -> 'exprs.Expr':
         func_name = self.name or self.fn.name
+        return tool_calls[func_name]['*'] >> self.__invoke_kwargs(R.args)
```

Comment:
Construct a JsonMapper instance here directly? Probably easier to read.

Reply:
Done.